### PR TITLE
Refactor Mysterion bootstrap method

### PR DESCRIPTION
**Pull request by donce**
This bootstrap method is very important, because application won't start if it works incorrectly, but it's very hard to understand what it does - I'm trying to improving readability of it.

### CODE REVIEW

File: src/app/mysterion.js [hunk 1]
```diff
@@ -45,35 +45,74 @@ class Mysterion {
   }
 
   async bootstrap () {
```
Comment (ignasbernotas) [inline src/app/mysterion.js]: I know it's a lot of work, but maybe we can combine forces and refactor it to use flow.

File: src/app/mysterion.js [hunk 2]
```diff
@@ -45,35 +45,74 @@ class Mysterion {
...
+
+  _getWindowSize (showTerms) {
+    if (showTerms) {
+      return this.config.windows.terms
+    } else {
+      return this.config.windows.app
+    }
+  }
+
+  _AreTermsAccepted () {
```
Comment (ignasbernotas) [inline src/app/mysterion.js]: "Are" -> "are"

File: src/app/mysterion.js [hunk 3]
```diff
@@ -45,35 +45,74 @@ class Mysterion {
...
+      return false
     }
+  }
+
+  _getSendFunction (browserWindow) {
+    return browserWindow.webContents.send.bind(browserWindow.webContents)
+  }
 
-    let browserWindow
+  _tryCreatingBrowserWindow () {
```
Comment (ignasbernotas) [inline src/app/mysterion.js]: I don't think this method needs to be prefixed with try if it still throws an exception on failure. I think createBrowserWindow is sufficient.

File: src/app/mysterion.js [hunk 4]
```diff
@@ -45,35 +45,74 @@ class Mysterion {
...
       console.error(e)
       this.bugReporter.captureException(e)
       throw new Error('Failed to open window.')
     }
+  }
 
-    const send = browserWindow.webContents.send.bind(browserWindow.webContents)
-    this.messageBus = new MainMessageBus(send, this.bugReporter.captureException)
-    this.communication = new MainCommunication(this.messageBus)
+  _tryCreatingWindow (size) {
```
Comment (ignasbernotas) [inline src/app/mysterion.js]: creatingWindow and creatingBrowserWindow naming is confusing, these could be merged into one?

File: src/app/mysterion.js (hunk 1 above)
Comment (donce) [reply src/app/mysterion.js]: Totally agree, but let's not increase scope of this PR, it's not healthy. After this is in `master`, we can work on that :)

File: src/app/mysterion.js (hunk 2 above)
Comment (donce) [reply src/app/mysterion.js]: Done

File: src/app/mysterion.js (hunk 3 above)
Comment (donce) [reply src/app/mysterion.js]: Done

File: src/app/mysterion.js (hunk 4 above)
Comment (donce) [reply src/app/mysterion.js]: Renamed to `createWindow` and `createBrowserWindow`. It is confusing, but at least it is consistent - i.e. we have `browserWindowFactory, windowFactory` properties in `Mysterion` class.

**Pull request merged**